cli/src/main.rs

```diff
@@ -52,6 +54,225 @@ pub fn main() {
+
+        let ident_raw = Command::new("git")
+            .args(["var", "GIT_AUTHOR_IDENT"])
+            .output()
+            .expect("problem ident. cannot invent!");
+
+        let ident = std::str::from_utf8(&ident_raw.stdout).expect("ident error");
+        // stand in for zingolabs.
+        if ident.starts_with("dannasessha <sessha@zingolabs.com>") {
+            // proceed
+        } else {
+            panic!("Zingo-cli's regtest mode must be run within its git tree");
+        }
+
+        // Cross-platform OsString
+        let mut worktree_home = OsString::new();
+        worktree_home.push(revparse.trim_end());
+
+        // convert this back into a path for windows compatibile dir building
+        let bin_location: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("bin"),
+        ]
+        .iter()
+        .collect();
+
+        let zcash_confs: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("conf"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut flagged_zcashd_conf: String = "--conf=".to_string();
+        flagged_zcashd_conf.push_str(zcash_confs.to_str().expect("error making zcash_datadir"));
+        flagged_zcashd_conf.push_str("zcash.conf");
+
+        let zcash_datadir: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("datadir"),
+            OsString::from("zcash"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut flagged_datadir: String = "--datadir=".to_string();
+        flagged_datadir.push_str(zcash_datadir.to_str().expect("error making zcash_datadir"));
+
+        let mut zcash_logs: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("logs"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut zcashd_bin = bin_location.to_owned();
+        zcashd_bin.push("zcashd");
+
+        // TODO reorg code, look for all needed bins ASAP
+        // check for file. This might be superfluous considering
+        // .expect() attached to the call, below?
+        if !std::path::Path::is_file(zcashd_bin.as_path()) {
+            panic!("can't find zcashd bin! exiting.");
+        }
+
+        println!("zcashd datadir: {}", &flagged_datadir);
+        println!("zcashd conf file: {}", &flagged_zcashd_conf);
+        let mut zcashd_command = Command::new(zcashd_bin)
+            .args([
+                "--printtoconsole",
+                &flagged_zcashd_conf,
+                &flagged_datadir,
+                // Right now I can't get zcashd to write to debug.log with this flag
+                //"-debuglogfile=.../zingolib/regtest/logs/debug.log",
```

Comment:
Are you using your `repo_root_name` variable to specify an absolute path?  `zcashd` probably refuses to operate on relative paths.

Also shouldn't the path name include `zcashd` to differentiate it from other kinds of logs?

Like:

`/home/redacted/zingolib/regtest/logs/zcashd/debug.log`

?
But also maybe we want to differentiate among `regtest` runs so maybe something like:

`/home/redacted/zingolib/regtest/logs/zcashd/$RUNSTARTTIME/debug.log`

?